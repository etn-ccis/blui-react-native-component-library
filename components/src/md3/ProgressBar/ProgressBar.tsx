import React from 'react';
import {
    MD3Theme,
    ProgressBar as PaperProgressBar,
    ProgressBarProps as PaperProgressBarProps,
    useTheme,
} from 'react-native-paper';
import { $DeepPartial } from '@callstack/react-theme-provider';

/**
 * ProgressBar component
 *
 * This component is a wrapper around the React Native Paper [ProgressBar](https://callstack.github.io/react-native-paper/progress-bar.html)
 * component. It accepts all the same props as the RNP component. The wrapper simply performs some minor theme / style overrides
 * in order to make the component look the way we want for Brightlayer UI projects.
 */

export type ProgressBarProps = Omit<PaperProgressBarProps, 'theme'> & {
    theme?: $DeepPartial<MD3Theme>;
};

export const ProgressBar: React.FC<ProgressBarProps> = (props) => {
    const { theme: themeOverride, ...otherProgressBarProps } = props;
    const theme = useTheme(themeOverride);

    return <PaperProgressBar {...otherProgressBarProps} theme={theme} />;
};
