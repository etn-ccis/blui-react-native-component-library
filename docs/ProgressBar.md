# Progress Bar

The `<ProgressBar>` component is a wrapper around React Native Paper's [ProgressBar](https://callstack.github.io/react-native-paper/docs/components/ProgressBar) component. It is used as a progress indicator of an activity. The wrapper simply performs some minor theme / style overrides in order to make the component look the way we want for Brightlayer UI projects.

<img width="100%" alt="Gradecomponent" src="./images/progressBar.png">

## Usage

```tsx
import { ProgressBar } from '@brightlayer-ui/react-native-components';

<ProgressBar progress={0.5} />;
```

### ProgressBar API

`ProgressBar` accepts all the same props as the React Native Paper's [ProgressBar](https://callstack.github.io/react-native-paper/docs/components/ProgressBar/#props) component.

### Styles

You can override the internal style used by Brightlayer UI by passing a `style` prop.
