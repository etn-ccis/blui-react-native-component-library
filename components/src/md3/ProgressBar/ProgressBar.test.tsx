import React from 'react';
import { ProgressBar } from './ProgressBar';
import { cleanup } from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';

describe('ProgressBar', () => {
    afterEach(cleanup);
    it('ProgressBar Renders', () => {
        const tree = TestRenderer.create(<ProgressBar progress={0.5} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
