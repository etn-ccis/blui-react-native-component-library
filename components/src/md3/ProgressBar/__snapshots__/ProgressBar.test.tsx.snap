// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar ProgressBar Renders 1`] = `
<View
  accessibilityRole="progressbar"
  accessibilityState={
    {
      "busy": true,
    }
  }
  accessibilityValue={
    {
      "max": 100,
      "min": 0,
      "now": 50,
    }
  }
  accessible={true}
  onLayout={[Function]}
  style={false}
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "rgba(231, 224, 236, 1)",
        "height": 4,
        "opacity": 0,
        "overflow": "hidden",
      }
    }
  />
</View>
`;
